Documentation/contributor-guide/dependency_management.md

```diff
@@ -1,95 +1,128 @@
 which depends on it should bump the dependency (`D1`), and then etcd just needs to bump `M1`. However, if the module (`M1`) somehow doesn't
-bump the problematic dependency, then etcd can still bump it (`D1`) directly following the same steps above. But as a long-term solution, etcd should 
+bump the problematic dependency, then etcd can still bump it (`D1`) directly following the same steps above. But as a long-term solution, etcd should
 try to remove the dependency on such module (`M1`) that lack maintenance.
 
 For mixed cases, in which some modules directly while others indirectly depend on a dependency, we have multiple options,
+
 - Bump the dependency for all modules, no matter it's direct or indirect dependency.
 - Bump the dependency only for modules which directly depend on it.
 
 We should try to follow the first way, and temporarily fall back to the second one if we run into any issue on the first way. Eventually we
 should fix the issue and ensure all modules depend on the same version of the dependency.
 
-## Known incompatible dependency updates
+### Known incompatible dependency updates
+
+#### arduino/setup-protoc
 
-### arduino/setup-protoc
 Please refer to [build(deps): bump arduino/setup-protoc from 1.3.0 to 2.0.0](https://github.com/etcd-io/etcd/pull/16016)
 
-## Rotation worksheet
+### Rotation worksheet
+
 The dependabot scheduling interval is weekly; it means dependabot will automatically raise a bunch of PRs per week.
 Usually human intervention is required each time. We have a [rotation worksheet](https://docs.google.com/spreadsheets/d/1DDWzbcOx1p32MhyelaPZ_SfYtAD6xRsrtGRZ9QXPOyQ/edit#gid=0),
 and everyone is welcome to participate; you just need to register your name in the worksheet.
 
-# Stable branches
+## Stable branches
+
 Usually we don't proactively bump dependencies for stable releases unless there are any CVEs or bugs that affect etcd.
 
 If we have to do it, then follow the same guidance above. Note that there is no `./scripts/fix.sh` in release-3.4, so no need to
 execute it for 3.4.
+
+## Golang versions
+
+The etcd project aims to maintain a development branch that is on the latest [Go version](https://go.dev/dl), ideally this will align with the Go version in use for Kubernetes project development. For an example on how to update etcd to a new minor release of Go refer to issue <https://github.com/etcd-io/etcd/issues/16393> and the linked pull requests.
+
+Suggested steps for performing a minor version upgrade for the etcd development branch:
+
+1. Carefully review new Go version release notes and blog for any deprecations, performance impacts or other considerations.
+2. Create a github issue to signal intent to upgrade and invite discussion, example <https://github.com/etcd-io/etcd/issues/16393>.
+3. Complete the upgrade locally in your development environment.
+4. Run performance benchmarks locally to compare before and after.
```

Comment:
Maybe we can mark it as optional or we should build benchmark pipeline or something to catch the diff, since we don't have basic test suite for this yet.

Reply:
Hmm I think given we don't have the automation that it should be mandatory to look at this manually for now?  I might be misunderstanding.

Reply:
> we don't have the automation

Yes. So, I am thinking that before given some basic test cases (for example, using benchmark tool to run some commands to collect the QPS metric somethings like that), the optional maybe better than mandatory. And sometimes, we just bump the version to fix CVE issues and we have to. The mandatory might bring the burden to contributors. Just my two cents. 

The change looks good to me overall.

Reply:
Thanks for the comment.

Reply:
Think that "run performance benchmark" is too obscure, we can leave it here for now, but I would like to file an issue to define a official performance validation for etcd so can stop depending on Kubernetes scalability to qualify performance.

Reply:
Yes I had to do a bit of guessing when I ran some benchmarks for comparing 1.19 with 1.20. It would certainly be good to have an agreed/defined approach in place that can be re-used, either manual trigger or automatic ci workflow would be my preference but even just an agreed set of instructions would be a start just so future contributors don't have to guess.

Reply:
Raised https://github.com/etcd-io/etcd/issues/16467